Makefile

```diff
@@ -1,19 +1,23 @@
-SHELL = /bin/bash
+SHELL             := /bin/bash
+.SHELLFLAGS       += -e -u -o pipefail
 
-export NAME       ?= mailserver-testing:ci
-export IMAGE_NAME := $(NAME)
+export IMAGE_NAME := mailserver-testing:ci
+export NAME       ?= $(IMAGE_NAME)
 
 # -----------------------------------------------
-# --- Generic Build Targets ---------------------
+# --- Generic Targets ---------------------------
 # -----------------------------------------------
 
 all: lint build backup generate-accounts tests clean
 
 build:
-	@ DOCKER_BUILDKIT=1 docker build --tag $(NAME) \
+	@ DOCKER_BUILDKIT=1 docker build . --tag $(NAME) \
 		--build-arg VCS_VERSION=$(shell git rev-parse --short HEAD) \
-		--build-arg VCS_REVISION=$(shell cat VERSION) \
-		.
+		--build-arg VCS_REVISION=$(shell cat VERSION)
+
```

Comment:
Why was the `.` shifted into before args? I had a quick look and didn't see any better ways to [set the context in the `docker build` docs](https://docs.docker.com/engine/reference/commandline/build/) other than the `.` :\

If this is what another maintainer prefers, I won't argue the change. I usually expect it at the end after options personally, and would have just preferred bumping `--tag` to a newline instead:

```Makefile
build:
	@ DOCKER_BUILDKIT=1 docker build \
	    --tag $(NAME) \
		--build-arg VCS_VERSION=$(shell git rev-parse --short HEAD) \
		--build-arg VCS_REVISION=$(shell cat VERSION) \
		.
```

Reply:
I did this to avoid the single dot on the new line, which looks weird to me. I think this might be a good compromise:

```make
build:
	@ DOCKER_BUILDKIT=1 docker build . \
	        --tag $(NAME) \
		--build-arg VCS_VERSION=$(shell git rev-parse --short HEAD) \
		--build-arg VCS_REVISION=$(shell cat VERSION)
```

Reply:
The man page synopsis mentions the build context as last argument. Not sure, how future proof the above will be. Also IMO we should avoid unusual syntax and revert them were we can instead of introducing new ones ;-)
Looking at you weird space in the shebang lines 😆

Reply:
Convinced 🙃